### ISSUE TITLE
Request for training log

### ISSUE THREAD

**Issue by Slyne** [espnet/espnet#5633]
**Describe your question**

Hi developers,
I was trying to run this experiment: https://github.com/espnet/espnet/tree/master/egs2/librispeech/asr2#expasr_train_discrete_asr_e_branchformer1_raw_wavlm_large_21_km2000_bpe_rm6000_bpe_ts5000_sp

But due to my GPU memory limit, I cannot run with 2 GPUs with `batch_bins: 768000000`. So I ran the experiment with 4 GPUs (16GB-V100). However, after 4 epochs, I begin to see nan loss and grad norm. So I was wondering if you happen to keep the training logs for this experiment? I wanna compare the training logs to try to locate the root cause of this issue.

Another request is that is it possible to add the kmeans model in the pretrained model: https://huggingface.co/espnet/simpleoier_ls960_asr2_train_e_branchformer1_raw_wavlm_large_21_km2000_bpe_rm6000_bpe_ts5000_sp ?  Just want to make sure everything is the same.

Thanks!

Attached the result after my 4 epoch training:
### WER

|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
|---|---|---|---|---|---|---|---|---|
|decode_ctc0.3_asr_model_valid.acc.ave/dev_clean|2703|54402|96.4|3.3|0.2|0.3|3.9|43.7|
|decode_ctc0.3_asr_model_valid.acc.ave/dev_other|2864|50948|94.0|5.5|0.5|0.5|6.5|53.0|
|decode_ctc0.3_asr_model_valid.acc.ave/test_clean|2620|52576|96.4|3.3|0.3|0.3|3.9|43.2|
|decode_ctc0.3_asr_model_valid.acc.ave/test_other|2939|52343|93.8|5.7|0.5|0.6|6.8|56.3|

### CER

|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
|---|---|---|---|---|---|---|---|---|
|decode_ctc0.3_asr_model_valid.acc.ave/dev_clean|2703|288456|99.0|0.6|0.5|0.3|1.4|43.7|
|decode_ctc0.3_asr_model_valid.acc.ave/dev_other|2864|265951|97.9|1.2|1.0|0.6|2.7|53.0|
|decode_ctc0.3_asr_model_valid.acc.ave/test_clean|2620|281530|99.0|0.5|0.5|0.3|1.3|43.2|
|decode_ctc0.3_asr_model_valid.acc.ave/test_other|2939|272758|97.9|1.1|1.0|0.6|2.8|56.3|

### TER

|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
|---|---|---|---|---|---|---|---|---|
|decode_ctc0.3_asr_model_valid.acc.ave/dev_clean|2703|68010|95.0|3.3|1.6|0.5|5.4|43.7|
|decode_ctc0.3_asr_model_valid.acc.ave/dev_other|2864|63110|92.0|5.8|2.2|1.0|9.0|53.0|
|decode_ctc0.3_asr_model_valid.acc.ave/test_clean|2620|65818|95.2|3.2|1.7|0.5|5.3|43.2|
|decode_ctc0.3_asr_model_valid.acc.ave/test_other|2939|65101|91.6|5.7|2.7|0.9|9.3|56.3|

**Comment by sw005320** [comment]
@simpleoier, can you respond to these requests?

**Comment by Slyne** [comment]
> Hi @Slyne , I may not have the log for the exact the same experiments. From the logs I have for similar librispeech/asr2 experiments, there is no nan loss or grad. Basic info: `A5000 x 2`, `batch_bins: 416000000`. The following is the logs for the first 4 epochs.
> 
> ```
> (trainer:737) INFO: 1epoch:train:1-1000batch: iter_time=6.825e-04, forward_time=0.143, loss_ctc=285.877, loss_att=237.454, acc=0.076, loss=251.981, backward_time=0.253, grad_norm=54.606, clip=100.000, loss_scale=6.554e+04, optim_step_time=0.046, optim0_lr0=1.337e-04, train_time=0.565
> (trainer:737) INFO: 1epoch:train:1001-2000batch: iter_time=1.512e-04, forward_time=0.142, loss_ctc=124.483, loss_att=162.661, acc=0.314, loss=151.208, backward_time=0.254, grad_norm=43.966, clip=100.000, loss_scale=6.554e+04, optim_step_time=0.046, optim0_lr0=4.004e-04, train_time=0.503
> (trainer:737) INFO: 1epoch:train:2001-3000batch: iter_time=1.435e-04, forward_time=0.142, loss_ctc=89.387, loss_att=91.472, acc=0.600, loss=90.847, backward_time=0.253, grad_norm=51.819, clip=100.000, loss_scale=1.311e+05, optim_step_time=0.046, optim0_lr0=6.671e-04, train_time=0.502
> (trainer:343) INFO: 1epoch results: [train] iter_time=2.794e-04, forward_time=0.142, loss_ctc=143.885, loss_att=139.836, acc=0.421, loss=141.051, backward_time=0.253, grad_norm=46.485, clip=100.000, loss_scale=9.782e+04, optim_step_time=0.046, optim0_lr0=5.260e-04, train_time=0.518, time=34 minutes and 6.14 seconds, total_count=3942, gpu_max_cached_mem_GB=22.949, [valid] loss_ctc=17.273, cer_ctc=0.060, loss_att=14.870, acc=0.902, cer=0.057, wer=0.707, loss=15.591, time=32.29 seconds, total_count=18, gpu_max_cached_mem_GB=22.949, [att_plot] time=37.61 seconds, total_count=0, gpu_max_cached_mem_GB=22.949
> (trainer:391) INFO: The best model has been updated: valid.acc
> (trainer:274) INFO: 2/35epoch started. Estimated time to finish: 20 hours, 1 minute and 3.12 seconds
> (trainer:737) INFO: 2epoch:train:1-1000batch: iter_time=5.951e-04, forward_time=0.142, loss_ctc=65.822, loss_att=53.906, acc=0.752, loss=57.481, backward_time=0.253, grad_norm=23.513, clip=98.900, loss_scale=2.545e+05, optim_step_time=0.047, optim0_lr0=0.001, train_time=0.563
> (trainer:737) INFO: 2epoch:train:1001-2000batch: iter_time=1.313e-04, forward_time=0.142, loss_ctc=62.064, loss_att=48.899, acc=0.773, loss=52.848, backward_time=0.253, grad_norm=16.994, clip=97.800, loss_scale=2.621e+05, optim_step_time=0.048, optim0_lr0=0.001, train_time=0.505
> (trainer:737) INFO: 2epoch:train:2001-3000batch: iter_time=1.224e-04, forward_time=0.141, loss_ctc=59.073, loss_att=45.344, acc=0.786, loss=49.463, backward_time=0.253, grad_norm=13.527, clip=96.100, loss_scale=5.091e+05, optim_step_time=0.047, optim0_lr0=0.002, train_time=0.504
> (trainer:343) INFO: 2epoch results: [train] iter_time=2.468e-04, forward_time=0.142, loss_ctc=61.013, loss_att=47.845, acc=0.776, loss=51.795, backward_time=0.253, grad_norm=16.361, clip=96.956, loss_scale=3.855e+05, optim_step_time=0.047, optim0_lr0=0.002, train_time=0.519, time=34 minutes and 10.05 seconds, total_count=7884, gpu_max_cached_mem_GB=23.316, [valid] loss_ctc=10.389, cer_ctc=0.034, loss_att=7.963, acc=0.941, cer=0.034, wer=0.550, loss=8.691, time=31.5 seconds, total_count=36, gpu_max_cached_mem_GB=23.316, [att_plot] time=37.11 seconds, total_count=0, gpu_max_cached_mem_GB=23.316
> (trainer:391) INFO: The best model has been updated: valid.acc
> (trainer:274) INFO: 3/35epoch started. Estimated time to finish: 19 hours, 26 minutes and 28.65 seconds
> (trainer:737) INFO: 3epoch:train:1-1000batch: iter_time=5.640e-04, forward_time=0.141, loss_ctc=54.518, loss_att=40.544, acc=0.802, loss=44.736, backward_time=0.252, grad_norm=9.627, clip=92.000, loss_scale=9.878e+05, optim_step_time=0.047, optim0_lr0=0.002, train_time=0.563
> (trainer:737) INFO: 3epoch:train:1001-2000batch: iter_time=1.228e-04, forward_time=0.141, loss_ctc=54.786, loss_att=40.273, acc=0.807, loss=44.627, backward_time=0.253, grad_norm=8.929, clip=93.000, loss_scale=1.049e+06, optim_step_time=0.047, optim0_lr0=0.003, train_time=0.504
> (trainer:737) INFO: 3epoch:train:2001-3000batch: iter_time=1.169e-04, forward_time=0.141, loss_ctc=53.224, loss_att=38.747, acc=0.811, loss=43.090, backward_time=0.253, grad_norm=8.309, clip=89.700, loss_scale=1.976e+06, optim_step_time=0.047, optim0_lr0=0.003, train_time=0.504
> (trainer:343) INFO: 3epoch results: [train] iter_time=2.315e-04, forward_time=0.141, loss_ctc=53.782, loss_att=39.407, acc=0.809, loss=43.719, backward_time=0.253, grad_norm=8.729, clip=91.350, loss_scale=1.519e+06, optim_step_time=0.047, optim0_lr0=0.003, train_time=0.519, time=34 minutes and 7.52 seconds, total_count=11826, gpu_max_cached_mem_GB=23.316, [valid] loss_ctc=8.834, cer_ctc=0.028, loss_att=6.570, acc=0.951, cer=0.028, wer=0.498, loss=7.249, time=31.79 seconds, total_count=54, gpu_max_cached_mem_GB=23.316, [att_plot] time=35.31 seconds, total_count=0, gpu_max_cached_mem_GB=23.316
> (trainer:391) INFO: The best model has been updated: valid.acc
> (trainer:274) INFO: 4/35epoch started. Estimated time to finish: 18 hours, 50 minutes and 37.53 seconds
> (trainer:737) INFO: 4epoch:train:1-1000batch: iter_time=7.410e-04, forward_time=0.142, loss_ctc=51.026, loss_att=36.573, acc=0.819, loss=40.909, backward_time=0.253, grad_norm=7.809, clip=88.400, loss_scale=3.829e+06, optim_step_time=0.047, optim0_lr0=0.003, train_time=0.561
> (trainer:737) INFO: 4epoch:train:1001-2000batch: iter_time=1.276e-04, forward_time=0.141, loss_ctc=51.268, loss_att=36.545, acc=0.821, loss=40.962, backward_time=0.252, grad_norm=7.999, clip=89.000, loss_scale=4.194e+06, optim_step_time=0.047, optim0_lr0=0.004, train_time=0.501
> (trainer:737) INFO: 4epoch:train:2001-3000batch: iter_time=1.181e-04, forward_time=0.141, loss_ctc=51.682, loss_att=36.661, acc=0.821, loss=41.167, backward_time=0.252, grad_norm=8.023, clip=90.100, loss_scale=7.659e+06, optim_step_time=0.047, optim0_lr0=0.004, train_time=0.499
> (trainer:343) INFO: 4epoch results: [train] iter_time=2.798e-04, forward_time=0.141, loss_ctc=51.054, loss_att=36.314, acc=0.821, loss=40.736, backward_time=0.252, grad_norm=8.040, clip=89.295, loss_scale=5.983e+06, optim_step_time=0.047, optim0_lr0=0.004, train_time=0.516, time=33 minutes and 55.66 seconds, total_count=15768, gpu_max_cached_mem_GB=23.316, [valid] loss_ctc=7.984, cer_ctc=0.025, loss_att=5.885, acc=0.955, cer=0.025, wer=0.460, loss=6.515, time=32.44 seconds, total_count=72, gpu_max_cached_mem_GB=23.316, [att_plot] time=38.32 seconds, total_count=0, gpu_max_cached_mem_GB=23.316
> ```
> 
> One reason I guess where the nan loss came from is the GPU type you're using. V100 may not support the half precision type `bfloat`. `float16` usually leads to nan loss or grad.
> 
> BTW, the kmeans model is in the huggingface repo [here](https://huggingface.co/espnet/simpleoier_ls960_asr2_train_e_branchformer1_raw_wavlm_large_21_km2000_bpe_rm6000_bpe_ts5000_sp/tree/main/exp/kmeans/wavlm_large_21_2000clusters). Are you able to use it?

Thanks! This is helpful.
I can try to train with fp32 first.
